Scrape calico-node metrics from tenant clusters

tuommaki opened the pull request:
Calico provides the networking layer for Giant Swarm tenant clusters and
therefore it's important to scrape metrics from it in order to be able to
configure proper monitoring for it.

Towards https://github.com/giantswarm/giantswarm/issues/6381

Commit:
Scrape calico-node metrics from tenant clusters

Calico provides the networking layer for Giant Swarm tenant clusters and
therefore it's important to scrape metrics from it in order to be able to
configure proper monitoring for it.

Towards https://github.com/giantswarm/giantswarm/issues/6381

Commit:
Merge branch 'master' into calico-node-metrics

JosephSalisbury reviewed (approved):
looks legit. i assume you get calico targets for control plane and tenant clusters etc?

tuommaki commented:
> looks legit. i assume you get calico targets for control plane and tenant clusters etc?

Control plane must be managed separately. I think control plane calico doesn't even expose metrics at the moment.

Commit:
Scrape calico-node PODs using pod service discovery role

This change adds support for scraping PODs directly with pod service discovery
role. This way there doesn't need to be service present for
prometheus-config-controller to see but prometheus discovers the PODs directly.
This is useful e.g. for calico-node PODs which don't have nor need any service
instance.

While adapting tests for this new service discovery config, the configuration
rendering test was now converted to be so called golden file test so
expectation string doesn't need to be maintained manually.

tuommaki committed:
Merge branch 'master' into calico-node-metrics

Commit:
keep calico-node metrics

Commit:
Merge branch 'calico-node-metrics' of ssh://github.com/giantswarm/prometheus-config-controller into calico-node-metrics

Commit:
remove calico-node job metric relabeling

Commit:
fix calico-node pod regex

Commit:
Distinct between POD w/ namespace & POD name matching

tuommaki commented:
Tested on `gauss`. Works now. Going.
